app/assets/javascripts/events.js.coffee

```diff
@@ -93,3 +93,15 @@ $ ->
           element: @
         )
         map.map.addOverlay(overlay)
+
+  if $('.toggle-past-events').length
+    $('.toggle-past-events').click (evt) =>
+      pastEvents = $('.past-events')
+      spanTxt = $('.toggle-past-events').children('span')
+
+      if pastEvents.hasClass('hidden')
+        pastEvents.removeClass('hidden')
+        spanTxt.text('Hide')
```

Comment:
You can use the ternary operation to set the text here
`pasEvents.hasClass('hidden') ? spanTxt.text('Hide') : spanTxt.text('Show')`